sqlx-core/src/postgres/type_info.rs

```diff
@@ -165,6 +165,11 @@ impl PgTypeInfo {
         self.0.kind()
     }
 
+    /// Returns the OID for this type.
+    pub fn oid(&self) -> Oid {
+        self.0.oid()
+    }
+
```

Comment:
This isn't a suitable way to expose this as the type OID may not be resolved yet, e.g. if constructed from `PgTypeInfo::with_name()`.

Reply:
Is there a suitable way of exposing the OID? If I want to deserialize a field to a different type depending on the column type, which could be the correct approach?